File: Java Word Examples/ProtectionApiDemo.java
import spire.cloud.word.sdk.client.*;
import spire.cloud.word.sdk.client.api.ProtectionApi;
import java.io.File;
import java.io.FileNotFoundException;

public class ProtectionApiDemo {
    static String appId = "your id";
    static String appKey = "your key";
    static String baseUrl = "https://api.e-iceblue.cn";
    static Configuration wordConfiguration = new Configuration(appId, appKey, baseUrl);
    static ProtectionApi protectionApi = new ProtectionApi(wordConfiguration);

        public static void protectDocument() throws ApiException {
            String fileName = "Template.docx";
            String name = fileName;
            String protectionType = "AllowOnlyReading"; //AllowOnlyComments,AllowOnlyFormFields,AllowOnlyRevisions,NoProtection
            String folder = "input";
            String storage = null;
            String docPassword = null;
            String protectiongPwd = "123456";
            String destFilePath = "output/Template_output.docx" ;
            protectionApi.protectDocument(name, protectionType, destFilePath, folder, storage, docPassword, protectiongPwd);
        }
        public static void protectDocumentWithFile() throws FileNotFoundException, ApiException {
            String inputFilePath = "D:/inputFile/Template.docx";
            File inputFile= new File(inputFilePath);
            String protectionType = "AllowOnlyReading"; //AllowOnlyComments,AllowOnlyFormFields,AllowOnlyRevisions,NoProtection
            String docPassword = null;
            String protectiongPwd = "123456";
            File response = protectionApi.protectDocumentInRequest(inputFile, protectionType, docPassword, protectiongPwd);
        }
}

File: C# Word Examples/ProtectionApiDemo.cs
using Spire.Cloud.Word.Sdk.Client;
using Spire.Cloud.Word.Sdk.Api;
using System;
using System.IO;

namespace CloudWord
{
    class ProtectionApiDemo
    {
        static string appId = "your id";
        static string appKey = "your key";
        static string basePath = "https://api.e-iceblue.cn";
        static Configuration wordConfiguration = new Configuration(appId, appKey, basePath);
        static ProtectionApi protectionApi = new ProtectionApi(wordConfiguration);
        public static void ProtectDocument()
        {
            var fileName = "Template.docx";
            string name = fileName;
            string protectionType = "AllowOnlyReading"; //AllowOnlyComments,AllowOnlyFormFields,AllowOnlyRevisions,NoProtection
            string folder = "input";
            string storage = null;
            string docPassword = null;
            string protectiongPwd = "123456";
            string destFilePath = "output/Template_output.docx" ;
            protectionApi.ProtectDocument(name, protectionType, destFilePath, folder, storage, docPassword, protectiongPwd);
        }
        public static void ProtectDocumentWithFile()
        {
            string inputFilePath = "D:/inputFile/Template.docx";
            System.IO.Stream inputFile = new FileStream(inputFilePath, FileMode.Open);
            string protectionType = "AllowOnlyReading"; //AllowOnlyComments,AllowOnlyFormFields,AllowOnlyRevisions,NoProtection
            string docPassword = null;
            string protectiongPwd = "123456";
            var response = protectionApi.ProtectDocumentInRequest(inputFile, protectionType, docPassword, protectiongPwd);
        }

    }
}